File: source/WeddingManager.Services/ExpenseService.cs
using System.Collections.Generic;
using WeddingManager.Core.Data;
using WeddingManager.Core.Repositories;
using WeddingManager.Core.Services;

namespace WeddingManager.Services
{
    public class ExpenseService : IExpenseService
    {
        public ExpenseService(IExpenseRepository expenseRepository)
        {
            _expenseRepository = expenseRepository;
        }

        public int CreateExpense(int companyId, Expense expense)
        {
            return _expenseRepository.CreateExpense(companyId, expense);
        }

        public IEnumerable<Expense> RetrieveExpenses(int companyId)
        {
            return _expenseRepository.RetrieveExpenses(companyId);
        }

        public void UpdateExpense(Expense expense)
        {
            _expenseRepository.UpdateExpense(expense);
        }

        public void DeleteExpense(int expenseId)
        {
            _expenseRepository.DeleteExpense(expenseId);
        }

        private readonly IExpenseRepository _expenseRepository;
    }
}

File: source/WeddingManager.Web/js/services/expenseService.js
(function(){
    angular.module('weddingManager')
    .service('expenseService', function(expenseFactory, appStateService){
        var service = {};
        var _expenses = [];
        
        service.createExpense = function(companyId, onSuccess, onError){
            var expense = {
                CreatedDate : new Date(),
                Description : "New Expense"
            };
            expenseFactory.createExpense(companyId, expense, function(expenseId){
                expense.Id = expenseId;
                onSuccess(expense);
            }, onError);
        }
        
        service.refreshExpenses = function(companyId, onSuccess, onError){
            expenseFactory.retrieveExpenses(companyId, function(expenses){
                _expenses = expenses;
                onSuccess(_expenses);
            }, onError);
        }
        
        service.updateExpense = function(expense, onSuccess, onError){
            expenseFactory.updateExpense(expense, function(){
                onSuccess();
            }, onError);
        }
        
        service.deleteExpense = function(expenseId, onSuccess, onError){
            expenseFactory.deleteExpense(expenseId, function(){
                appStateService.setExpense(null);
                onSuccess();
            }, onError);
        }
        
        return service;
    });
})();